### PR TITLE
QGM: Add support for Get

### LINKED ISSUE #9136: HIR ⇒ QGM ⇒ MIR support for `Get`
- For alignment with existing notation, we will name the `BaseTable` box as `Get` instead.
- As part of this issue, one needs to refine the fake `Catalog` implementation, so that the planner can resolve fake tables in the unit tests (see the `test.rs` logic `query_model`).

### PR DESCRIPTION
**Pull request by wangandi**
Add support for `Get` in QGM.

Following the design doc, I figured this was the time to add the field `unique_keys` to the `QueryBox` struct.

Open question: the original design doc only had sample tables with a single one-column unique key. Does this look like good formatting when there are multiple sets of unique keys, some of which involve multiple columns?
basic-0011 dot

### Motivation

Closes #9136.

### Tips for Reviewers

As expected, a bunch of this PR is just test changes. There is some spillover to outside the `src/sql/src/querymodel` crate because I wanted to reuse code from other test crates:
* The method `generate_explanation` has been generalized to take any object that implements the `ExprHumanizer` trait so I can print MIRs with types in src/sql unit tests in order to confirm that keys have correctly been carried over from HIR to QGM and then MIR.
* I realized that instead of parsing a test command with custom code, like what's here: https://github.com/MaterializeInc/materialize/blob/cbcad02b3a0cedfe79843e173d9fa5a102327d14/src/expr-test-util/src/lib.rs#L192
  I could just create an enum that represents the test command and use lowertest::deserialize to construct the enum. For convenience when writing test commands, I modified the DSL to allow strings to be constructed without quotes as long as the string is not "null". 

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by asenac** [comment]
That formatting looks good to me. Alternatively, we could add a separate line prefixed with `UNIQUE KEY` for each one (this what I used in https://github.com/asenac/rust-sql for example), but this one is perhaps more compact. We could try this one and revisit it later in #9180.

File: src/sql/src/query_model/mod.rs [hunk 1]
```diff
@@ -149,7 +153,9 @@ pub enum BoxType {
 }
 
 #[derive(Debug)]
-pub struct BaseTable {/* @todo table metadata from the catalog */}
+pub struct Get {
+    id: expr::GlobalId,
```
Comment (asenac) [inline src/sql/src/query_model/mod.rs:157]: Eventually, we'll need to add the RelationType here too, when we implement the `typ` method in `BoxScalarExpr`.

Comment (asenac) [reply src/sql/src/query_model/mod.rs:157]: We could add it as a follow up task and add an option to the dot generator to dump types.

**Review by asenac — APPROVED** [review]
Nice!

File: src/sql/src/query_model/mod.rs (hunk 1 above)
Comment (wangandi) [reply src/sql/src/query_model/mod.rs:157]: Should types just included in the dot graph always? Or is the concern that if a column is something like record type, the record type can be very long?

Comment (wangandi) [reply src/sql/src/query_model/mod.rs:157]: Added #9601 as a follow-up issue.

**Pull request merged**